cubeviz/tools/collapse_cube.py

```diff
@@ -311,187 +228,321 @@ def hide_start_end(self, dohide):
+            self.ui.start_label.setEnabled(False)
+            self.ui.start_example_label.setEnabled(False)
+            self.ui.start_input.setEnabled(False)
+            self.ui.end_label.setEnabled(False)
+            self.ui.end_example_label.setEnabled(False)
+            self.ui.end_input.setEnabled(False)
+        else:
+            self.ui.start_label.setEnabled(True)
+            self.ui.start_example_label.setEnabled(True)
+            self.ui.start_input.setEnabled(True)
+            self.ui.end_label.setEnabled(True)
+            self.ui.end_example_label.setEnabled(True)
+            self.ui.end_input.setEnabled(True)
+
+    def _calculate_callback_wavelength_checks(self, start_wavelength, end_wavelength):
+
+        log.debug('_calculate_callback_wavelength_checks with start {} and end {}'.format(
+            start_wavelength, end_wavelength))
+
+        if len(start_wavelength) == 0:
+            self.ui.end_label.setStyleSheet("color: rgba(0, 0, 255, 128)")
+            self.ui.error_label.setText('No start wavelength, assuming last spectral wavelength.')
+            start_wavelength = self.wavelengths[0]
+
+        if len(end_wavelength) == 0:
+            self.ui.end_label.setStyleSheet("color: rgba(0, 0, 255, 128)")
+            self.ui.error_label.setText('No end wavelength, assuming last spectral wavelength.')
+            end_wavelength = self.wavelengths[-1]
+
+        try:
+            start_wavelength = float(start_wavelength)
+        except:
+            self.ui.start_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
+            self.ui.error_label.setText('Start wavelength is not a floating point number.')
+            self.ui.error_label.setVisible(True)
+
+            return None, None
+
+        try:
+            end_wavelength = float(end_wavelength)
+        except:
+            self.ui.start_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
+            self.ui.error_label.setText('End wavelength is not a floating point number.')
+            self.ui.error_label.setVisible(True)
+
+            return None, None
+
+        if start_wavelength < self.wavelengths[0]:
+            self.ui.start_label.setStyleSheet("color: rgba(0, 0, 255, 128)")
+            self.ui.error_label.setText('Start wavelength is out of range, setting to end point.')
+            self.ui.error_label.setVisible(True)
+
+            start_wavelength = self.wavelengths[0]
+            
+        if start_wavelength > self.wavelengths[-1]:
+            self.ui.start_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
+            self.ui.error_label.setText('Start wavelength is out of range.')
+            self.ui.error_label.setVisible(True)
+
+            start_wavelength = None
+            
+
+        if end_wavelength > self.wavelengths[-1]:
+            self.ui.end_label.setStyleSheet("color: rgba(0, 0, 255, 128)")
+            self.ui.error_label.setText('End wavelength is out of range, setting to end point.')
```

Comment:
I set `Start Wavelength` to 1000 and `End Wavelength` to 10, and this is the error message I get, which is not what I would expect. I think the `start_wavelength < end_wavelength` test should take priority over these?

Reply:
I switched the order to see the start> end message.